fix(image-gen): include thoughtSignature in Gemini image generation response

Chesars opened the pull request:
fix(image-gen): include thoughtSignature in Gemini image generation response

  ## Relevant issues

  Fixes #17184

  ## Pre-Submission checklist

  - [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory
  - [ ] I have added a screenshot of my new test passing locally
  - [ ] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
  - [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

  ## Type

  🐛 Bug Fix

  ## Changes

  ### Summary

  Gemini 3 Pro image preview model returns a `thoughtSignature` field in the response that is required for interactive image editing workflows. This field was being discarded during response transformation.

  ### Solution

  - Added optional `thought_signature` field to `ImageObject` class
  - Updated response transformation in both `gemini/` and `vertex_ai/` providers to extract and include `thoughtSignature` from the API response
  - Added unit test to verify `thought_signature` is correctly extracted
  
  ### Backward Compatibility

  This change is fully backward compatible - the new field is optional

Chesars committed:
fix(image-gen): add thought_signature to ImageObject for Gemini 3 Pro

Fixes #17184 - Gemini 3 Pro image preview model returns a thoughtSignature
field required for interactive image editing. This change:

- Adds thought_signature field to ImageObject class
- Updates Gemini and Vertex AI transformations to extract thoughtSignature
- Adds test for thought_signature in response transformation

krrishdholakia commented on litellm/llms/gemini/image_generation/transformation.py:
```diff
@@ -232,6 +232,7 @@ def transform_image_generation_response(
                             model_response.data.append(ImageObject(
                                 b64_json=inline_data["data"],
                                 url=None,
+                                thought_signature=part.get("thoughtSignature"),
```
we should avoid having this be a root level param - as it's not openai compatible. 

can we nest this inside a field called `provider_specified_fields` - similar to chat completions

Chesars committed:
refactor(image-gen): move thought_signature to provider_specific_fields

Per review feedback, thought_signature should not be a root-level
param on ImageObject as it's not OpenAI compatible. Moved to
provider_specific_fields dict to match the pattern used in chat
completions (Message, Delta, Choices, etc).
